backend/tests/v5/e2e/routes/teamspaces/projects/containers/metadata.test.js

```diff
@@ -0,0 +1,226 @@
+			expect(res.body.code).toEqual(templates.projectNotFound.code);
+		});
+
+		test('should fail if the container does not exist', async () => {
+			const res = await agent.patch(`/v5/teamspaces/${teamspace}/projects/${project.id}/containers/${ServiceHelper.generateRandomString()}/metadata/${metadataId}?key=${users.tsAdmin.apiKey}`)
+				.expect(templates.containerNotFound.status);
+			expect(res.body.status).toEqual(templates.containerNotFound.status);
+		});
+
+		test('should fail if the container is a federation', async () => {
+			const res = await agent.patch(`/v5/teamspaces/${teamspace}/projects/${project.id}/containers/${federation._id}/metadata/${metadataId}?key=${users.tsAdmin.apiKey}`);
+			expect(templates.containerNotFound.status);
+			expect(res.body.status).toEqual(templates.containerNotFound.status);
+		});
+
+		test('should fail if the metadata do not exist', async () => {
+			const res = await agent.patch(`/v5/teamspaces/${teamspace}/projects/${project.id}/containers/${ServiceHelper.generateRandomString()}/metadata/${ServiceHelper.generateRandomString()}?key=${users.tsAdmin.apiKey}`)
+				.expect(templates.metadataNotFound.status);
+			expect(res.body.status).toEqual(templates.metadataNotFound.status);
+		});
+
+		test('should fail if the user does not have permissions to access the container', async () => {
+			const res = await agent.patch(`${route}?key=${users.viewer.apiKey}`)
+				.expect(templates.notAuthorized.status);
+			expect(res.body.status).toEqual(templates.notAuthorized.status);
+		});
+
+		test('should fail if the user is trying to update non custom metadata', async () => {
+			const res = await agent.patch(`${route}?key=${users.tsAdmin.apiKey}`)
+				.send({
+					metadata: [
+						{ key: nonCustomMetadata.key, value: ServiceHelper.generateRandomString() },
+						{ key: customMetadata.key, value: ServiceHelper.generateRandomString() },
+					],
+				}).expect(templates.invalidArguments.status);
+			expect(res.body.status).toEqual(templates.invalidArguments.status);
+		});
+
+		test('should fail if the user is tryign to add metadata with missing value', async () => {
+			const res = await agent.patch(`${route}?key=${users.tsAdmin.apiKey}`)
+				.send({
+					metadata: [{ key: nonCustomMetadata.key }],
+				}).expect(templates.invalidArguments.status);
+			expect(res.body.status).toEqual(templates.invalidArguments.status);
+		});
+
+		test('should add new metadata', async () => {
+			const metadataToAdd = {
+				key: ServiceHelper.generateRandomString(),
+				value: ServiceHelper.generateRandomString() };
+			await agent.patch(`${route}?key=${users.tsAdmin.apiKey}`).send({ metadata: [metadataToAdd] })
+				.expect(templates.ok.status);
+
+			const updatedMetdata = await agent.get(`${routeV4}?key=${users.tsAdmin.apiKey}`);
+			const updatedMetadataObj = updatedMetdata.body.meta[0].metadata;
+			const itemAdded = Object.prototype.hasOwnProperty.call(updatedMetadataObj, metadataToAdd.key)
```

Comment:
We need to check the whole document.

We want to make sure that it didn't remove other fields that it shouldn't have removed

Reply:
similar with the tests below